file-formats/ront/type/zif_aff_ront_v1.intf.abap

```diff
@@ -0,0 +1,71 @@
+INTERFACE zif_aff_ront_v1
+  PUBLIC.
+
+  CONSTANTS:
+    BEGIN OF co_category,
+      "! <p class="shorttext">Business Object</p>
+      "! Business object
+      business_object      TYPE c LENGTH 2 VALUE `BO`,
+      "! <p class="shorttext">Technical Object</p>
+      "! Technical object
+      technical_object     TYPE c LENGTH 2 VALUE `TO`,
+      "! <p class="shorttext">Analytical Object</p>
+      "! Analytical object
+      analytical_object    TYPE c LENGTH 2 VALUE `AO`,
+      "! <p class="shorttext">Configuration Object</p>
+      "! Configuration object
+      configuration_object TYPE c LENGTH 2 VALUE `CO`,
+    END OF co_category.
+
+  TYPES:
+    "! <p class="shorttext">Node</p>
+    BEGIN OF ty_rontnode,
+      "! <p class="shorttext">Name</p>
+      "! Node name
+      "! $required
+      node_name     TYPE c LENGTH 40,
+
+      "! <p class="shorttext">Root Node Flag</p>
+      "! Root node flag
+      "! $required
+      is_root       TYPE abap_bool,
+
+      "! <p class="shorttext">Name</p>
+      "! Raw (camel case) node name
+      "! $required
+      node_name_raw TYPE c LENGTH 40,
+
+      "! <p class="shorttext">Description</p>
+      "! Description
+      "! $required
+      description   TYPE zif_aff_types_v1=>ty_description_60,
+    END OF ty_rontnode.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    BEGIN OF ty_main,
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_60_cloud,
+
+      "! <p class="shorttext">Type Category</p>
+      "! Type category
+      "! $required
+      "! $values {@link zif_aff_ront_v1.data:co_category}
+      type_category  TYPE c LENGTH 2,
+
+      "! <p class="shorttext">Root SAP Object Node Type</p>
+      "! Root SAP object node type
+      "! $required
+      root_node_type TYPE ty_rontnode,
+
+      "! <p class="shorttext">SAP Object Node Types</p>
+      "! $required
```

Comment:
If you make the table required, then f.ex. in the example RONT an empty array is printed into the json when the node_types arrays would be empty:
"nodeTypes": [ ]

In my opinion it is sufficient to have the fields of structure ty_rontnode on required.

Reply:
I agree with @huber-nicolas as long empty `nodeTypes` are allowed.

Reply:
Thanks for pointing this out. A SAP Object Type with a root SAP Object Node Type only, that is with empty nodeTypes is allowed. I will remove the $required ABAP Doc comment for the table nodeTypes.